gtsam/hybrid/DiscreteBoundaryFactor.cpp

```diff
@@ -0,0 +1,104 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ *  @file DiscreteBoundaryFactor.cpp
+ *  @date Mach, 2025
+ *  @author Varun Agrawal
+ */
+
+#include <gtsam/hybrid/DiscreteBoundaryFactor.h>
+
+namespace gtsam {
+
+/* ************************************************************************ */
+DecisionTreeFactor DiscreteFactorFromErrors(
+    const DiscreteKeys& discreteKeys,
+    const AlgebraicDecisionTree<Key>& errors) {
+  double min_log = errors.min();
+  AlgebraicDecisionTree<Key> potentials(
+      errors, [&min_log](const double x) { return exp(-(x - min_log)); });
+
+  return DecisionTreeFactor(discreteKeys, potentials);
+}
+
+/* ************************************************************************ */
+DecisionTreeFactor DiscreteBoundaryFactor::ComputeDiscreteBoundary(
+    const DiscreteKeys& dkeys,
+    const HybridNonlinearFactor::FactorValuePairs& factors,
+    const gtsam::Values& values) {
+  auto calculateError = [&](const NonlinearFactorValuePair& pair) -> double {
+    if (pair.first) {
+      auto gaussianNoiseModel = std::dynamic_pointer_cast<noiseModel::Gaussian>(
+          pair.first->noiseModel());
```

Comment:
I vaguely remember doing some weird things to handle different noise model cases. Will this "just work" even if you were using a Cauchy noise model or something?

Reply:
Unfortunately, I don't think so. We haven't really considered robust noise models in our scheme yet, but that's something we hope to accomplish soon!

Reply:
I believe all we need is a way to define `logDetR` for RobustNoiseModel and that should allow us to use `noiseModel::Base` as the parent class.